### ISSUE TITLE
terminating with uncaught exception of type std::bad_cast: std::bad_cast

### ISSUE THREAD

**Issue by BlindGalaxy4817** [ptitSeb/box64#240]
I'm trying to run a Satisfactory dedicated server on ARM64 (4 cores 24gb of RAM). I'm using box64 and box86 to boot the server. Whenever I do, it crashes with this error message: " terminating with uncaught exception of type std::bad_cast: std::bad_cast". I was told to put this here, sorry if it's in the wrong spot.

**Comment by ptitSeb** [comment]
That's fine, you are in the right place (I asked you to create the ticket here :) )

How did you install that server? using steamcmd or something else?

**Comment by BlindGalaxy4817** [comment]
I installed it with steamcmd running on box86, then ran it using box64.

**Comment by tastysauce** [comment]
FWIW, I'm running into the same issue

**Comment by QuintenQVD0** [comment]
I am also having this issue, hope my logs can provice help to wy it doesn't work and how this can be fixt

https://hastebin.com/izisilonax.md

https://hastebin.com/bojajowuki.md

**Comment by A-iko** [comment]
I'm trying to pull the same stunt, and I'm dealing with the same error.

For reproduction:

- Create a free server on Oracle Cloud
(4 Ampere aarch64 cores + 24GB, Ubuntu Canonical)
- Install box64 and box86 using the custom repos
- Move the Satisfactory Dedicated Server to the server (I couldn't get SteamCMD to run, but it's not required)
- Run (I use `BOX86_NOBANNER=1 ./FactoryServer.sh` just in case)

Let me know if there's anything I can do to help and thank you for your hard work!

**Comment by QuintenQVD0** [comment]
some documentiation I found arround this error
- https://en.cppreference.com/w/cpp/types/bad_cast
- https://stackoverflow.com/questions/19960496/compiles-but-getting-uncaught-exception-of-type-stdbad-cast
-

**Comment by bennesp** [comment]
Same error, following this guide to setup an environment: https://github.com/Raezroth/Linux-ARM-Gaming-Chroot/blob/master/README.md . Then installed SteamCMD, downloaded the game and started in the chroot env. Crashing with

```
terminating with uncaught exception of type std::bad_cast: std::bad_cast                                                                                                               
Signal 6 caught.                                                                                                                                                                       
Malloc Size=65538 LargeMemoryPoolOffset=65554                                                                                                                                          
CommonUnixCrashHandler: Signal=6                                                                                                                                                       
[2022.11.20-14.55.58:525][  0]LogCore: === Critical error: ===                                                                                                                         
Unhandled Exception: SIGABRT: abort() called                                                                                                                                           
                                                                                                                                                                                       
[2022.11.20-14.55.58:526][  0]LogCore: Fatal error!                                                                                                                                    
                                                                                                                                                                                       
0x0000ffff87d87642 libUE4Server-Core-Linux-Shipping.so!FUnixPlatformStackWalk::CaptureStackBackTrace(unsigned long long*, unsigned int, void*) [D:/ws/SB-220602145101-842/UE4/Engine/So
urce/Runtime/Core/Private/Unix/UnixPlatformStackWalk.cpp:693]                                                                                                                          
                                                                                                                                                                                       
[2022.11.20-14.55.58:538][  0]LogExit: Executing StaticShutdownAfterError   
```

**Comment by parzival-space** [comment]
Any updates on this? I basically want to do the same.

**Comment by ptitSeb** [comment]
Is it still happening?

**Comment by QuintenQVD0** [comment]
> Is it still happening?

yes it is

**Comment by ptitSeb** [comment]
Mmmm, ok. I'll install Satisfactory on my arm test machine and will give a it try

**Comment by QuintenQVD0** [comment]
note for my that is I am trying to run the server not the game itself

**Comment by ptitSeb** [comment]
Yeah ok. There is a separate download on steam for the Dedicated server, so I'll use this

**Comment by ptitSeb** [comment]
So, I did try and indeed I get Segfault.
After some analysis, t's a cycling lib dependancly: A use B, and B use A.
Not sure how I will handle that.

**Comment by parzival-space** [comment]
How did you identify this? Just out of curiosity. I am not a C expert

**Comment by ptitSeb** [comment]
That's not the whole story. (I found at using gdb, it was obvious there).
Once I workaround this recursive lib stuff, I actually get the `std::bad_cast: std::bad_cast` message. So, I need to check what's wrong there too.

**Comment by NickGildenstern** [comment]
Any updates ?

**Comment by ptitSeb** [comment]
I did some debugging, but didn't found anything conclusive. I'm still unsurre why it happens, I need to speed more time on this.

**Comment by NickGildenstern** [comment]
Is there any possible way how we could help ?

**Comment by ptitSeb** [comment]
Not sure. The exception occurs during the initialisation of lib. I really haven't found enough detail yet. I suspect it's an issue with the elfloader / librarian, with somehow an issue with a wrong symbol being used. But I really need to put more time in this to really get the symbol being executed that launch the exception. It's a dynamic value, not a static one, so I need to analyse (unless someone with debugging knowledge and good understanding of the ELF structure and runtime linking want to take over)

**Comment by NickGildenstern** [comment]
No updates right ?

**Comment by rajdakin** [comment]
No update, maybe it works now, but Satisfactory (the full game) doesn't work (can't reach the menu, it crashes while loading).
